pkg/util/date.go

```diff
@@ -724,3 +724,14 @@ func ConvertToPST(t time.Time) (*time.Time, error) {
 
 	return &pt, nil
 }
+
+// GetLocDateFromState gets the time based on the given state's timezone
+func GetLocDateFromState(state string) (*time.Time, error) {
+	tzString := StandardTimeZoneForState(state)
```

Comment:
@SudipKatwal  please note that the state name should be in abbreviation form, in case the state is not found it defaults to CST timezone. Please add that to the description of the function.